api/v1beta2/foundationdbbackup_types.go

```diff
@@ -322,6 +322,11 @@ func (backup *FoundationDBBackup) CheckReconciliation() (bool, error) {
 		reconciled = false
 	}
 
+	if isRunning && backup.BackupURL() != backup.Status.BackupDetails.URL {
```

Comment:
I think it would be better to create a dedicated method (e.g. `NeedsConfigurationChange`) for this and combine the `SnapshotPeriodSeconds` and URL check. That way the method can be extended in the future without adding more if statements in CheckReconciliation.